x-pack/plugins/timelines/public/plugin.ts

```diff
@@ -43,6 +45,13 @@ export class TimelinesPlugin implements Plugin<void, TimelinesUIStart> {
           const state = getState();
           if (state && state.app) {
             this._store = undefined;
+          } else {
+            if (props.onTGridStateChange) {
+              this._storeUnsubscribe = this._store.subscribe(
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                throttle(() => props.onTGridStateChange!(getState()), 500)
```

Comment:
Is the `!` a TS feature? I would expect the optional chaining operator `?.` here.

Reply:
`!` is to tell TS that a value is definitely not null. In this case, we already did an `if (props.onTGridStateChange)` check, but for some reason that's not good enough for Typescript here? So I had to add this `!` operator to assure the compiler that everything is fine.

You'll notice I had to disable the linting rule:
```
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Usually we don't want to use this `!` operator if it's avoidable, but in this case there must be some deeper architectural issues going on if the surrounding `if` block isn't reassuring the compiler that everything is okay.

Reply:
Ok and now I just tried removing the `!` and it's no longer erroring anymore???? What are computers what is typescript I don't know what's going on

Reply:
LOL NOPE it broke again but my VSCode didn't realize it until after I already merged and pushed